src/index/Vocabulary.cpp

```diff
@@ -37,22 +45,10 @@ bool Vocabulary<StringType, ComparatorType, IndexT>::PrefixRanges::contain(
 }
 
 // _____________________________________________________________________________
+
 template <class S, class C, typename I>
-void Vocabulary<S, C, I>::readFromFile(const string& fileName) {
-  LOG(INFO) << "Reading vocabulary from file " << fileName << " ..."
-            << std::endl;
-  vocabulary_.close();
-  vocabulary_.open(fileName);
-  if constexpr (std::is_same_v<S, detail::UnderlyingVocabRdfsVocabulary>) {
-    const auto& internalExternalVocab =
-        vocabulary_.getUnderlyingVocabulary().getUnderlyingVocabulary();
-    LOG(INFO) << "Done, number of words: "
-              << internalExternalVocab.internalVocab().size() << std::endl;
-    LOG(INFO) << "Number of words in external vocabulary: "
-              << internalExternalVocab.externalVocab().size() << std::endl;
-  } else {
-    LOG(INFO) << "Done, number of words: " << vocabulary_.size() << std::endl;
-  }
+void Vocabulary<S, C, I>::readFromFile(const string& filename) {
+  vocabulary_.getUnderlyingVocabulary().readFromFile(filename);
```

Comment:
I have a similar refactoring in the polymorphic vocab pr, so we we won't have many conflicts there.